bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/navigation/graph/TabNavGraph.kt

```diff
@@ -0,0 +1,37 @@
+package network.bisq.mobile.presentation.ui.navigation.graph
+
+import PriceRepository
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.composable
+import androidx.navigation.navigation
+import network.bisq.mobile.domain.data.repository.BisqStatsRepository
+import network.bisq.mobile.presentation.ui.navigation.Routes
+import network.bisq.mobile.presentation.ui.navigation.Graph
+import network.bisq.mobile.presentation.ui.screens.*
+
+fun NavGraphBuilder.TabNavGraph(
+    rootNavController: NavHostController,
+    innerPadding: PaddingValues
+) {
+    navigation(
+        startDestination = Routes.TabHome.name,
+        route = Graph.MainScreenGraph
+    ) {
+        composable(route = Routes.TabHome.name) {
+            val presenter = GettingStartedPresenter(priceRepository = PriceRepository(), bisqStatsRepository = BisqStatsRepository())

```

Comment:
Refactor: UI shouldn't know which presenter they are using, just rely on the interface they defined (as you have correctly done in other cases! :) )

always receive presenter by parameter.

This is for many reasons:

 - UI as dumb as possible
 - do not hardcode dependencies and do not depend on implementations, rather on interfaces (for testing and maintainability purposes)

Reply:
Understood. But some confusions here.

For `SplashScreen` and `OnBoardingScreen`, I have instantiated the presenter inside the screen, which is wrong.

For `GettingStartedScreen`, I have instantiated the presenter, where we define Nav routes. And pass the presenter to `GettingStartedScreen` as a param.
So we can dynamically instantiate some other object in the future as pass it along and the `View` won't even know it.

Few Qs:
1. If not in the routing page, where should be presenters be instantiated?
2. I guess, this is where the 'koin' library plays it's role?

Once I get clarity on this, I will apply the same to all the 3 screens.

Reply:
we have addressed this on matrix, and with a recent merge that buddha himself reviewed so I think this is answered? Let me know if we need to clarify anyrthing else :muscle:

Reply:
Ha ha. It's clear now. Thanks.